src/i22_bluesky/util/settings.py

```diff
@@ -2,51 +2,74 @@
 from pathlib import Path
 
 import bluesky.plan_stubs as bps
-from ophyd_async.core import Device
-from ophyd_async.epics.areadetector import (
-    NDAttributeDataType,
-    NDAttributesXML,
-    PilatusDetector,
-)
+from dodal.devices.linkam3 import Linkam3
+from ophyd_async.core import StandardDetector
+from ophyd_async.epics.areadetector import NDAttributeDataType, NDAttributesXML
+from ophyd_async.epics.areadetector.drivers import ADBase
+
+SAVES_ROOT = Path(__file__).parent.parent.parent
+
+
+def get_device_save_dir(plan_name: str) -> Path:
+    return SAVES_ROOT / f"pvs/{plan_name}"
+
+
+def get_ad_xml_dir(plan_name: str) -> Path:
+    return SAVES_ROOT / f"xml/{plan_name}"
 
 
 @lru_cache(maxsize=1)
-def make_stats_sum_xml(path: Path) -> str:
+def _enable_stats_sum_xml() -> NDAttributesXML:
     xml = NDAttributesXML()
     xml.add_param(
         "StatsTotal",
         "TOTAL",
         NDAttributeDataType.DOUBLE,
         description="Sum of each detector frame",
     )
-    path.write_text(str(xml))
-    return str(path)
+    return xml
 
 
 @lru_cache(maxsize=1)
-def make_saxs_linkam_stamping_xml(linkam: Device, path: Path):
+def _enable_pv_stamping(name: str, pv: str, description: str) -> NDAttributesXML:
     xml = NDAttributesXML()
-    pv = linkam.temp.source.split("://")[1]
-    xml.add_epics_pv(
-        "Temperature",
-        f"{pv}",
-        description="Current linkam temperature",
+    xml.add_epics_pv(name, pv, description=description)
+    return xml
+
+
+@lru_cache(maxsize=1)
+def path_stats_sum_enabled(base_path: Path) -> str:
+    path = base_path / "stats_sum_stamping.xml"
+    path.write_text(str(_enable_stats_sum_xml()))
+    return str(path)
+
+
+@lru_cache(maxsize=1)
+def path_temp_stamping_enabled(linkam: Linkam3, base_path: Path):
+    path = base_path / "drv_linkam_stamping.xml"
+    path.write_text(
+        str(
+            _enable_pv_stamping(
+                "Temperature",
+                linkam.temp.source.split("://")[1],
+                "Current linkam temperature",
+            )
+        )
     )
-    path.write_text(str(xml))
     return str(path)
 
 
-def load_saxs_linkam_settings(linkam: Device, saxs: PilatusDetector, path: Path):
+def stamp_temp_pv(linkam: Linkam3, stamped_detector: StandardDetector, base_path: Path):
+    assert isinstance(driver := stamped_detector.drv, ADBase)
```

Comment:
please explain why walrus operator here

Reply:
As in what it's doing or the motivation of doing it?

```suggestion
   driver = stamped_detector.drv
    assert isinstance(driver, ADBase)
```

^ is what it's doing, and using the walrus makes it slightly shorter.

Since the PV stamping requires the AreaDetector plugin chain to work, this isinstance check ensures that the detector that is being stamped is an AD

Reply:
ok